bindings/wasm/src/account/wasm_account.rs

```diff
@@ -0,0 +1,320 @@
+        .update_document_unchecked(document_copy)
+        .await
+        .map(|_| JsValue::undefined())
+        .wasm_result()
+    })
+  }
+
+  /// Fetches the latest changes from the tangle and **overwrites** the local document.
+  ///
+  /// If a DID is managed from distributed accounts, this should be called before making changes
+  /// to the identity, to avoid publishing updates that would be ignored.
+  #[wasm_bindgen(js_name = fetchState)]
+  pub fn fetch_state(&mut self) -> Promise {
+    let account = self.0.clone();
+    future_to_promise(async move {
+      account
+        .as_ref()
+        .borrow_mut()
+        .fetch_state()
+        .await
+        .map(|_| JsValue::undefined())
+        .wasm_result()
+    })
+  }
+}
+
+impl From<Account> for WasmAccount {
+  fn from(account: Account) -> WasmAccount {
+    WasmAccount(Rc::new(WasmRefCell::new(account)))
+  }
+}
+
+#[wasm_bindgen]
+extern "C" {
+  #[wasm_bindgen(typescript_type = "Promise<Credential>")]
+  pub type PromiseCredential;
+
+  #[wasm_bindgen(typescript_type = "Promise<Presentation>")]
+  pub type PromisePresentation;
+
+  #[wasm_bindgen(typescript_type = "Promise<Document>")]
+  pub type PromiseDocument;
+}
+
+#[wasm_bindgen]
+extern "C" {
+  #[wasm_bindgen(typescript_type = "PublishOptions")]
+  pub type WasmPublishOptions;
+
+  #[wasm_bindgen(getter, method)]
+  pub fn forceIntegrationUpdate(this: &WasmPublishOptions) -> Option<bool>;
+
+  #[wasm_bindgen(getter, method)]
+  pub fn signWith(this: &WasmPublishOptions) -> Option<String>;
+}
+
+#[wasm_bindgen(typescript_custom_section)]
+const TS_PUBLISH_OPTIONS: &'static str = r#"
+/**
+ * Options to customize how identities are published to the Tangle.
+**/
+export type PublishOptions = {
+    /**
+     * Whether to force the publication to be an integration update.
+     * If this option is not set, the account automatically determines whether
+     * an update needs to be published as an integration or a diff update.
+     * Publishing as an integration update is always valid, but not recommended
+     * for identities with many updates.
+     *
+     * See the IOTA DID method specification for more details.
+     */
+     forceIntegrationUpdate?: boolean,
+
+
+    /**
+     *
+     *
+     * Set the fragment of a verification method with which to sign the update.
+     * This must point to an Ed25519 method with a capability invocation
+     * verification relationship.
+     */
+     signWith?: string
+ }
+"#;
+
+impl From<WasmPublishOptions> for PublishOptions {
+  fn from(publish_options: WasmPublishOptions) -> Self {
+    let mut options: PublishOptions = PublishOptions::new();
+
+    if let Some(force_integration) = publish_options.forceIntegrationUpdate() {
+      options = options.force_integration_update(force_integration);
+    }
+
+    if let Some(sign_with) = publish_options.signWith() {
+      let sign_with: String = sign_with;
+      options = options.sign_with(sign_with);
+    };
```

Comment:
Is there some weird coercion happening here, this can be condensed right?

```suggestion
    if let Some(sign_with) = publish_options.signWith() {
      options = options.sign_with(sign_with);
    };
```

Reply:
No, I think I wanted to write something like 
```rust 
if let Some(sign_with: String) = publish_options.signWith()
```
But apparently this doesn't work in Rust. I removed the redundant line.